CMakeLists.txt

```diff
@@ -0,0 +1,24 @@
+cmake_minimum_required(VERSION 3.19)
+
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD_REQUIRED on)
+
+if (APPLE)
+    foreach (HOMEBREW_PKG openssl libressl sqlite)
+        execute_process(COMMAND brew --prefix ${HOMEBREW_PKG} OUTPUT_VARIABLE HOMEBREW_PREFIX OUTPUT_STRIP_TRAILING_WHITESPACE)
+        list(APPEND CMAKE_PREFIX_PATH "${HOMEBREW_PREFIX}")
+    endforeach ()
+endif ()
+
+find_package(OpenSSL REQUIRED)
+find_package(SQLite3 REQUIRED)
```

Comment:
Automatically finds these libraries. It supports a wide range of environments and OS, which makes thinking about special library paths a thing of the past.